### PR TITLE
feat(components): add star rating

### PR DESCRIPTION
**Pull request by livgrhm**
### What is accomplished by your PR?

Create new star rating component as suggested in: https://github.com/vuesion/vuesion/issues/192

WIP!
Need to add full test coverage & look at greater flexibility/customisability for the component

### Is there something controversial in your PR?

please take a moment to tell us about your choices...

### Link to the Issue

or a link to a ticket system...

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR
- [ ] Change documentation for the code introduced by your PR
- [x] Add Story / Design System Page for a new component introduced by your PR

### CODE REVIEW

File: src/app/shared/components/VueStarRating/VueStarRating.vue [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      this.numSelectedStars = this.selectedNumStars;
+    }
+  },
+};
+</script>
+
+<style lang="scss" module>
+@import '~@/app/shared/design-system';
+
+$star-inactive-color: $brand-text-secondary-color-inverse;
```
Comment (devCrossNet) [inline src/app/shared/components/VueStarRating/VueStarRating.vue]: This should be pulled out into a variables file before finalizing the PR

File: src/app/shared/components/icons/VueIconStar/VueIconStar.vue [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<template>
```
Comment (devCrossNet) [inline src/app/shared/components/icons/VueIconStar/VueIconStar.vue]: Add to icons story

Comment (livgrhm) [reply src/app/shared/components/icons/VueIconStar/VueIconStar.vue]: 👍

File: src/app/shared/components/VueStarRating/VueStarRating.vue (hunk 1 above)
Comment (livgrhm) [reply src/app/shared/components/VueStarRating/VueStarRating.vue]: 👍

**Comment by livgrhm** [comment]
@devCrossNet in the example provided in the issue, there are a fair few variations on the component (https://github.com/vuesion/vuesion/issues/192) like half stars, all the way to 'glowing' stars.

Do you have a view on which of these are useful to include? Otherwise could merge this as a v1 & improve later :)

**Comment by devCrossNet** [comment]
I think half stars would be good for displaying, not necessarily for selection.

Simple use-case one user gives 5 stars, another one gives 3 - display would be 4,5 stars.

I like the idea of the visual feedback on click, maybe @einfalles has some ideas for this component, too?

**Pull request merged**